File: backend/app/schemas/prompt.py
import uuid
from datetime import datetime

from pydantic import BaseModel, Field

from app.models.enums import PromptCategory


# ── Request schemas ──────────────────────────────────────────────

class PromptCreate(BaseModel):
    title: str = Field(min_length=1, max_length=200)
    description: str = Field(default="", max_length=2000)
    content: str = Field(min_length=1, max_length=50000)
    category: PromptCategory


class PromptUpdate(BaseModel):
    title: str | None = Field(default=None, min_length=1, max_length=200)
    description: str | None = Field(default=None, max_length=2000)
    content: str | None = Field(default=None, min_length=1, max_length=50000)
    category: PromptCategory | None = None
    change_summary: str = Field(default="", max_length=500)
    lock_version: int


# ── Response schemas ─────────────────────────────────────────────

class PromptVersionResponse(BaseModel):
    id: uuid.UUID
    prompt_id: uuid.UUID
    version_number: int
    content: str
    change_summary: str
    created_at: datetime

    model_config = {"from_attributes": True}


class EvaluationSummaryInPrompt(BaseModel):
    id: uuid.UUID
    overall_score: float | None
    status: str
    created_at: datetime

    model_config = {"from_attributes": True}


class PromptResponse(BaseModel):
    id: uuid.UUID
    title: str
    description: str
    category: PromptCategory
    is_active: bool
    lock_version: int
    created_at: datetime
    updated_at: datetime
    version_count: int

    model_config = {"from_attributes": True}


class PromptDetailResponse(PromptResponse):
    current_version: PromptVersionResponse
    latest_evaluation: EvaluationSummaryInPrompt | None = None


class PromptListResponse(BaseModel):
    items: list[PromptResponse]
    total: int
    page: int
    page_size: int

File: frontend/src/types/prompt.ts
export enum PromptCategory {
  ORCHESTRATOR = "orchestrator",
  TASK_EXECUTION = "task_execution",
}

export interface PromptVersion {
  id: string;
  prompt_id: string;
  version_number: number;
  content: string;
  change_summary: string;
  created_at: string;
}

export interface Prompt {
  id: string;
  title: string;
  description: string;
  category: PromptCategory;
  is_active: boolean;
  lock_version: number;
  created_at: string;
  updated_at: string;
  version_count: number;
}

export interface PromptDetail extends Prompt {
  current_version: PromptVersion;
  latest_evaluation?: {
    id: string;
    overall_score: number | null;
    status: string;
    created_at: string;
  };
}

export interface PromptListResponse {
  items: Prompt[];
  total: number;
  page: number;
  page_size: number;
}

export interface PromptCreate {
  title: string;
  description?: string;
  content: string;
  category: PromptCategory;
}

export interface PromptUpdate {
  title?: string;
  description?: string;
  content?: string;
  category?: PromptCategory;
  change_summary?: string;
  lock_version: number;
}
